src/server/identity/server/dex_api_test.go

```diff
@@ -215,7 +291,46 @@ func TestUpdateConnector(t *testing.T) {
 				Name:          "newname",
 				Type:          "mockPassword",
 				ConfigVersion: 3,
-				JsonConfig:    `{"username": "user", "password": "pass"}`,
+				Config:        &types.Any{Value: []byte(`{"password":"pass","username":"user"}`)},
+			},
+		},
+		{
+			req: &identity.IDPConnector{
+				Id:            "conn",
+				Type:          "mockPassword",
+				Config:        &types.Any{Value: []byte("---\nusername: user\npassword: pass")},
+				ConfigVersion: 4,
+			},
+			expected: &identity.IDPConnector{
+				Id:            "conn",
+				Name:          "newname",
+				Type:          "mockPassword",
+				ConfigVersion: 4,
+				Config:        &types.Any{Value: []byte(`{"password":"pass","username":"user"}`)},
```

Comment:
it would be nice if the return value also came back in YAML. I don't think that's as critical though, and could be done in a separate PR